syntax = "proto2";

package efficient_paper;

message PaperInfo {
    optional Paper paper = 1;
    optional Publication pub = 2;
    optional Code code = 3;
    optional Note note = 4;
    optional Keyword keyword = 5;
    optional Cover cover = 6;
}

message Paper {
    optional string title = 1 [default="paper title"];
    optional string abbr = 2 [default=""];
    optional string url = 3;
    repeated string authors = 4;
    repeated string institutions = 5;
}

message Publication {
    optional string where = 1 [default="arXiv"];
    optional int32 year = 2;
}

message Code {
    optional string type = 1 [default="Pytorch"];
    optional string url = 2;
}

message Note {
    optional string url = 1 [default="note.md"];
}

message Cover {
    optional string url = 1;
}

message Keyword {
    repeated Word words = 1;
    enum Word {
        none = 1;
        overlap = 2;
        attention_sparsity = 3;
        activation_sparsity = 4;
        modeling = 5;
        structured_sparsity = 6;
        sparse_pruning = 7;        
        quantization = 8;
        survey = 9;
        low_rank = 10;
        fusion = 11;
        tool = 12;
        kv_cache = 13;
        efficient_training = 14;
        structure_design = 15;
        weight_sparsity = 16;
        deployment = 17;
        // TODO: please add your key words here.
    }
}

message SearchWord {
    repeated string background_words = 1;
    repeated string key_words = 2;
    repeated string exclude_words = 3;
}